addons/web/static/src/js/fields/field_utils.js

```diff
@@ -132,7 +132,7 @@ function formatDate(value, field, options) {
  * @returns {string}
  */
 function formatDateTime(value, field, options) {
-    if (value === false) {
+    if (value === false || value === undefined) {
```

Comment:
fix is wrong I think.  The correct way is to prevent undefined value to enter the basic model in the first place.  I would suggest to change something with the makedefaultrecord (or the postprocess?) method, to make sure that date/datetime fields are initialized to false.